draft-specifications/Model.md

```diff
@@ -526,69 +472,56 @@ local peer, the sent and received values will have the same type.
+the sent and received values will be Equal.
 
-OCapN holds equality invariant for all [Atoms](#atom) and
-[Containers](#container) passed between peers.
+> OCapN Equal values may be locally distinguishable by oother operators
+> depending on the language implementation and type.
 
 > In JavaScript, none of `==`, `===`, or `Object.is` are sufficient to compare
-> equality of values, but a composite operator that takes the OCapN type for
-> every value into account.
+> Equality of values, which instead must take into account the OCapN type for
+> each value.
 > For example, Float64 can be compared by `Object.is`, which preserves the
-> distinction between 0 and -0 and non-equivalence of any NaN to any other NaN,
-
-## Pass Invariant Identity
-
-Values of a type have an identity if a value of that type is identical to
-itself and no other value.
-A type holds identity invariant over passage between OCapN peers if values of that
-type can be passed from a local peer to a remote peer, then returned to the local peer,
-and the value that was sent is identical the the value received _and identical
-to no other value_.
-
-> The operators necessary for checking identify may vary for each language and
-> even for types in one language.
-> Pass Invariant Identity obliges OCapN to track a value's identity until the value
-> becomes unreachable.
-
-OCapN holds identity invariant only for values of type [Target](#target).
-
-OCapN does not hold identity invariant for values of type [Promise](#promise).
-
-OCapN does not guarantee identity is invariant for any values of [Atom](#atom)
-or [Container](#container) types.
+> distinction between 0 and -0 and lack of distinction between any two NaN
+> values.
+> `Object.is` is also sufficient for Target provided that the OCapN
+> implementation holds invariant a single object identity for every Target.
+> The JavaScript ArrayBuffer representation of an OCapN ByteArray must be
+> compared byte-for-byte.
 
-> Notably, in JavaScript, all Atomc types have trivial Pass Invariant Identity
-> except the JavaScript `ArrayBuffer` for an OCapN [ByteArray](#bytearray) and
-> the JavaScript object representation of an OCapN [Selector](#selector).
-> OCapN does not obligate a JavaScript implementation to intern a single identity
-> for all equivalent ByteArrays or Selectors.
 
 # JSON Invariants
 
 OCapN holds invariant that:
 
-- Any JSON document that a JavaScript peer can produce using
+- Any JSON text that a JavaScript peer can produce using
   `JSON.stringify`,
-- excluding those that contain any Unicode surrogates,
+- excluding those in which the Unicode representation of any object member name
+  or string includes a surrogate,
 - can then be read back with `JSON.parse`,
 - sent to any other OCapN peer,
 - returned from any OCapN peer,
-- then saved back to an identical JSON document with `JSON.stringify`.
+- then saved back to an identical JSON text with `JSON.stringify`.
```

Comment:
```suggestion
- then saved back to an equivalent JSON text with `JSON.stringify`.
```

Even though `JSON.stringify` is much more deterministic than the JSON standard demands, it is not canonical. It much encode the field of the same struct in a different order.